lib/AuthService.ts

```diff
@@ -0,0 +1,148 @@
+    }
+  }
+
+  async logout(options?: any): Promise<void> {
+    let redirectUri = null;
+    options = options || {};
+    if (typeof options === 'string') {
+      redirectUri = options;
+      // If a relative path was passed, convert to absolute URI
+      if (redirectUri.charAt(0) === '/') {
+        redirectUri = window.location.origin + redirectUri;
+      }
+      options = {
+        postLogoutRedirectUri: redirectUri
+      };
+    }
+    await this.sdk.signOut(options);
+  }
+
+  async loginRedirect(additionalParams?: object): Promise<void> {
+    const { scopes, responseType } = this.sdk.options;
+    const params = Object.assign({
+      scopes: scopes || ['openid', 'email', 'profile'],
+      responseType: responseType || ['id_token', 'token']
+    }, additionalParams);
+
+    return this.sdk.token.getWithRedirect(params);
+  }
+
+  async handleAuthentication(): Promise<void> {
+    const { tokens } = await this.sdk.token.parseFromUrl();
+    if (tokens.idToken) {
+      this.sdk.tokenManager.add(ID_TOKEN_STORAGE_KEY, tokens.idToken);
+    }
+    if (tokens.accessToken) {
+      this.sdk.tokenManager.add(ACCESS_TOKEN_STORAGE_KEY, tokens.accessToken);
+    }
+  }
+
+  setFromUri(fromUri?: string): void {
+    // Use current location if fromUri was not passed
+    fromUri = fromUri || window.location.href;
+    // If a relative path was passed, convert to absolute URI
+    if (fromUri.charAt(0) === '/') {
+      fromUri = window.location.origin + fromUri;
+    }
+    sessionStorage.setItem(REFERRER_PATH_STORAGE_KEY, fromUri);
+  }
+
+  getFromUri(relative: boolean = false): string {
+    let fromUri = sessionStorage.getItem(REFERRER_PATH_STORAGE_KEY) || window.location.origin;
+    sessionStorage.removeItem(REFERRER_PATH_STORAGE_KEY);
+    if (!relative) {
+      return fromUri;
+    }
+
+    const url = new URL(fromUri);
+    fromUri = `${url.pathname}${url.search}${url.hash}`
+    return fromUri;
+  }
+
+  getTokenManager(): TokenManager {
+    return this.sdk.tokenManager;
+  }
+
+  // Angular specific APIs
+
+  isAuthenticated(): Promise<boolean> {
+    const authState = this.sdk.authStateManager.getAuthState();
+    return Promise.resolve(authState.isAuthenticated);
+  }
+
+  // React specific APIs
+
+  redirect(additionalParams?: object): Promise<void> {
+    return this.loginRedirect(additionalParams);
+  }
+
+  getAuthState(): AuthState {
+    return this.sdk.authStateManager.getAuthState();
+  }
+
+  async updateAuthState(): Promise<void> {
+    const accessToken = await this.sdk.tokenManager.get(ACCESS_TOKEN_STORAGE_KEY) as AccessToken;
+    const idToken = await this.sdk.tokenManager.get(ID_TOKEN_STORAGE_KEY) as IDToken;
+    this.sdk.authStateManager.updateAuthState({ accessToken, idToken });
+  }
+
+  on(eventName: String, callback: Function): Function {
+    this.sdk.emitter.on(eventName, callback);
+    return () => this.sdk.emitter.off(eventName);
+  }
+}
+
+export default AuthService;
```

Comment:
I saw downstream SDKs are using `buildConfigObject` from `@okta/configuration-validation` to transform configs. Then the logic can still keep the transformation logic when implicitly initial `OktaAuth` in downstream SDKs.
But users are not supposing to initial this `AuthService` by themselves.

Reply:
I think when we make the switch to using auth-js as the service we will also leave validation of the config to auth-js. There will be no call to `buildConfigObject`, unless auth-js is making that call. `buildConfigObject` has already been reduced to do only a few little things for backward compatibility, all of which we can deprecate/remove in a major version bump.

Reply:
I think this can be a discussion on how we want to change configs in the downstream SDKs. 
For the react sdk, it can be:
1. Explicitly provide `oktaAuth` in props. Then the config can just be the exactly config for `auth-js`;
2. Implicitly initial `oktaAuth` in <Security>. Then, it the question for react-sdk to decide if the props should be changed or not.

Reply:
I believe we want auth-js to own its config. There may be some extra props on <Security> that are only used by React, but if the prop is config meant for the implicitly created `oktaAuth`, it should be in the same exact format that auth-js expects, so we can link to the auth-js documentation for reference.

The only things I see `buildConfigObject` doing are:

- Allow `scope` as a string instead of `scopes` as an array 
- Allow `autoRenew` and `storage` token manager options at top-level
- Allow `response_type` or `responseType` as a string instead of `responseType` as an array 
- Allow `client_id` instead of `clientId`

In all these cases I think we can deprecate/remove the option or implement support directly in auth-js.

Reply:
Checked the readme from `react-sdk`, looks like all props are in camelCase now, and **NO** autoRenew from top-level. I think it's safe to remove the `@okta/configuration-validation` dependency from `react-sdk`. We just need to be careful when make downstream SDKs changes.